src/data/validate.rs

```diff
@@ -0,0 +1,365 @@
+use crate::config::Mapping;
+
+/// Validate paths for entity
+pub fn validate_paths_for_entity(
+    entity: &str,
+    mappings: &Vec<Mapping>,
+    api_schema: &serde_json::Map<String, serde_json::Value>,
+) -> anyhow::Result<()> {
+    // if entity name is not set in api_schema throw an exception
+    if !api_schema.contains_key(entity) {
+        anyhow::bail!("Entity {} not found in API schema", entity);
+    }
+
+    for entry in mappings {
+        let path_mapping = match entry {
+            Mapping::ByPath(path_mapping) => path_mapping,
+            Mapping::ByScript(_) => continue,
+        };
+
+        let path = path_mapping.entity_path.split('.').collect::<Vec<_>>();
+
+        // if path starts with ? its optional
+        if path[0].ends_with('?') {
```

Comment:
As you need `path[0]` in multiple places it might be a good idea to introduce a variable for it (as this access isn't free / can potentially panic everywhere)